### ISSUE TITLE
Inverted "flipX()" and "flipY()"

### ISSUE THREAD

**Issue by Fennec-hub** [thednp/svg-path-commander#13]
[svg-path-commander.js#L252](https://github.com/thednp/svg-path-commander/blob/a5b2bc9753b1476a9b165aedb41bd22a80db9c43/src/svg-path-commander.js#L252)

`flipX()` and `flipY()` should be inverted, `flipX()` is flipping the `y` coordinates and `flipY()`  the `x` coordinates ... Not a big deal, but worth mentioning.

Great library by the way, I'm using it to create a small SVG path editor so big thanks.

**Comment by thednp** [comment]
The logic is [correct](https://github.com/thednp/svg-path-commander/blob/a5b2bc9753b1476a9b165aedb41bd22a80db9c43/src/svg-path-commander.js#L252-L268):

```js 
  flipX() {
    this.transform({ rotate: [180, 0, 0] }); // first value corresponds to X
    return this;
  }
```

```js
  flipX() {
    this.transform({ rotate: [0, 180, 0] }); // second value corresponds to Y
    return this;
  }
```

**Comment by thednp** [comment]
> Great library by the way, I'm using it to create a small SVG path editor so big thanks.

Link it in.

**Comment by Fennec-hub** [comment]
If by flipping you mean rotating, then yes the logic is correct, but most people think of flipping as mirroring and you can see this behavior in a lot of graphic libraries like `fabric.js`, `konva.js`, `three.js` ... so the logic would be more of a reverse scale.

```js
 flipX() {
    this.transform({ scale: [-1, 1, 1] }); 
    return this;
 }
 
 flipY() {
    this.transform({ scale: [1, -1, 1] }); 
    return this;
 }
``` 

> Link it in.

It's part of an ongoing project, the SVG editor part will look more or less like this [demo here](https://jxnblk.github.io/paths) ... I'll keep you in touch when the project is done.

**Comment by Fennec-hub** [comment]
I just noticed that the `transform` method don't handle all scale axes so in this case just inverting the two methods work fine.


```js
  flipX() {
    this.transform({ rotate: [0, 180, 0] }); // Rotate the Y axis, mirror the X axis
    return this;
  }
  flipY() {
    this.transform({ rotate: [180, 0, 0] }); // Rotate the X axis, mirror the Y axis
    return this;
  }
```

**Comment by thednp** [comment]
> I just noticed that the `transform` method don't handle all scale axis so in this case just inverting the two methods works fine.

Yes it [does](https://github.com/thednp/svg-path-commander/blob/a5b2bc9753b1476a9b165aedb41bd22a80db9c43/src/process/getSVGMatrix.js#L51-L53). Perhaps we have a bug?

**Comment by Fennec-hub** [comment]
Yes it does and it works ... please don't mind my last post.

**Comment by thednp** [comment]
@Fennec-hub you are right about flipX / flipY, our original source in SVGPath was like that and I changed it at some point.

I remember I was working with font-icons -> SVG icons and it didn't make sense to me I couldn't flip icons upside-down because I was confused on why flipX did not return the expected outcome.

I just checked the native DOMMatrix interface and it does exactly the same as you expected, so...

The future version will fix that.
